server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -80,6 +90,33 @@ public TransportRolloverAction(
         this.rolloverService = rolloverService;
         this.client = client;
         this.activeShardsObserver = new ActiveShardsObserver(clusterService, threadPool);
+        this.rolloverTaskExecutor = (currentState, tasks) -> {
```

Comment:
Can we turn this lambda into an explicit class? It makes it easier to reason about the state that it uses (should not depend on local state).